x-pack/plugins/alerting/server/task_runner/action_scheduler/action_scheduler.ts

```diff
@@ -74,9 +74,13 @@ export class ActionScheduler<
     this.schedulers.sort((a, b) => a.priority - b.priority);
   }
 
-  public async run(
-    alerts: Record<string, Alert<State, Context, ActionGroupIds | RecoveryActionGroupId>>
-  ): Promise<RunResult> {
+  public async run({
```

Comment:
I didn't make them optional because we always pass the active and recovered alerts. getProcessAlerts returns `{}` incase of no alerts.

The default values (activeCurrentAlerts = {}`) are for the unit tests actually. Some tests does not pass any alerts at all and `Object.values(activeCurrentAlerts)` breaks the execution as `activeCurrentAlerts` is undefined.

I can remove the default fields and use `Object.values(activeCurrentAlerts || {})` as well

Reply:
I think maybe I'm more confused - we have tests passing `undefined` for some of these properties - are they bypassing the typing somehow?  Because it _looks_ to me like they are required. 

Seems like we should clean this up, but we can do that in a separate PR - can you create an issue?